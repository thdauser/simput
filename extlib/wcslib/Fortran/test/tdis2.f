*=======================================================================
*
* WCSLIB 7.7 - an implementation of the FITS WCS standard.
* Copyright (C) 1995-2021, Mark Calabretta
*
* This file is part of WCSLIB.
*
* WCSLIB is free software: you can redistribute it and/or modify it
* under the terms of the GNU Lesser General Public License as published
* by the Free Software Foundation, either version 3 of the License, or
* (at your option) any later version.
*
* WCSLIB is distributed in the hope that it will be useful, but WITHOUT
* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
* License for more details.
*
* You should have received a copy of the GNU Lesser General Public
* License along with WCSLIB.  If not, see http://www.gnu.org/licenses.
*
* Author: Mark Calabretta, Australia Telescope National Facility, CSIRO.
* http://www.atnf.csiro.au/people/Mark.Calabretta
* $Id: tdis2.f,v 7.7 2021/07/12 06:36:49 mcalabre Exp $
*=======================================================================

      PROGRAM TDIS2
*-----------------------------------------------------------------------
*
* TDIS2 tests the WCSLIB Fortran routines for manipulating the contents
* of a disprm struct.  Two disprm structs are used.  One, residing in a
* linprm struct, is referenced by "address" (DISP), and the other (DISL)
* resides locally in an INTEGER array.
*
* DISP and WCSP, which are meant to hold addresses, are declared as
* INTEGER arrays of length 2 to accomodate 64-bit machines for which
* sizeof(void *) = 2*sizeof(int).
*
* Test input comes from TPV7.fits.
*-----------------------------------------------------------------------
      INTEGER   DPMAX
      PARAMETER (DPMAX = 256)

      LOGICAL   GOTEND
      INTEGER   DISP(2), IBLOCK, IKEYRC, J, JK, JN, K, NAXIS, NDP,
     :          NKEYRC, NREJECT, NWCS, STATL, STATP, STATUS, WCSP(2)
      DOUBLE PRECISION MAXDIS(2), TOTDIS
      CHARACTER DTYPE(2)*72, KEYREC*80, HEADER*28801, INFILE*9

*     On some systems, such as Sun Sparc, the structs MUST be aligned
*     on a double precision boundary, done here using an equivalence.
*     Failure to do this may result in mysterious "bus errors".
      INCLUDE 'wcserr.inc'
      INCLUDE 'wcshdr.inc'
      INCLUDE 'wcs.inc'
      INCLUDE 'lin.inc'
      INCLUDE 'dis.inc'
      INTEGER   DP(DPMAX*DPLEN), WCS(WCSLEN), LIN(LINLEN), DISL(DISLEN)
      DOUBLE PRECISION DUMMY1, DUMMY2, DUMMY3, DUMMY4
      EQUIVALENCE (DUMMY1, DP)
      EQUIVALENCE (DUMMY2, WCS)
      EQUIVALENCE (DUMMY3, LIN)
      EQUIVALENCE (DUMMY4, DISL)

      DATA INFILE /'TPV7.fits'/
*-----------------------------------------------------------------------
      STATUS = WCSERR_ENABLE (1)

      WRITE (*, 10)
 10   FORMAT (
     :  'Testing WCSLIB Fortran disprm access routines (tdis2.f)',/,
     :  '-------------------------------------------------------')

*     Open the FITS WCS test header for formatted, direct I/O.
      OPEN (UNIT=1, FILE=INFILE, FORM='FORMATTED', ACCESS='DIRECT',
     :      RECL=80, IOSTAT=STATUS)
      IF (STATUS.NE.0) THEN
        WRITE (*, 20) STATUS, INFILE
 20     FORMAT ('ERROR',I3,' opening ',A)
        GO TO 999
      END IF

*     Read in the FITS header, excluding COMMENT and HISTORY keyrecords.
      K = 1
      NKEYRC = 0
      GOTEND = .FALSE.
      DO 50 IBLOCK = 0, 100
        DO 40 IKEYRC = 1, 36
          READ (1, '(A80)', REC=36*IBLOCK+IKEYRC, IOSTAT=STATUS) KEYREC
          IF (STATUS.NE.0) THEN
            WRITE (*, 30) STATUS
 30         FORMAT ('ERROR',I5,' reading header.')
            GO TO 999
          END IF

          IF (KEYREC(:8).EQ.'        ') GO TO 40
          IF (KEYREC(:8).EQ.'COMMENT ') GO TO 40
          IF (KEYREC(:8).EQ.'HISTORY ') GO TO 40

          HEADER(K:) = KEYREC
          K = K + 80
          NKEYRC = NKEYRC + 1

          IF (KEYREC(:10).EQ.'END       ') THEN
*           An END keyrecord was read, but read the rest of the block.
            GOTEND = .TRUE.
          END IF
 40     CONTINUE

        IF (GOTEND) GO TO 60
 50   CONTINUE

 60   CLOSE (UNIT=1)


*     Parse the header.
      CALL FLUSH(6)
      STATUS = WCSPIH (HEADER, NKEYRC, WCSHDR_none, 2, NREJECT, NWCS,
     :                 WCSP)
      IF (STATUS.NE.0) GO TO 999

*     Copy into our WCSPRM struct.
      STATUS = WCSVCOPY (WCSP, 0, WCS)
      IF (STATUS.NE.0) GO TO 999

*     WCSSET translates the TPV "projection" into a sequent distortion.
      STATUS = WCSSET (WCS)
      IF (STATUS.NE.0) THEN
        STATUS = WCSPERR (WCS, CHAR(0))
        GO TO 999
      END IF

*     Make a shallow copy of the linprm struct.
      STATUS = WCSGTI (WCS, WCS_LIN, LIN)
      IF (STATUS.NE.0) THEN
        STATUS = WCSPERR (WCS, CHAR(0))
        GO TO 999
      END IF

*     Extract a "pointer" to the disprm struct.
      STATUS = LINGTI (LIN, LIN_DISSEQ, DISP)
      IF (STATUS.NE.0) THEN
        STATUS = LINPERR (LIN, CHAR(0))
        GO TO 999
      END IF

*     Add a little texture.  The first argument of DISPTD (and DISPRT,
*     DISGET, etc.) is unity to indicate that DISP is a "pointer".
      STATP = DISPTD (1, DISP, DIS_MAXDIS, 2.0010D0, 1, 0)
      IF (STATP.NE.0) GO TO 998
      STATP = DISPTD (1, DISP, DIS_MAXDIS, 2.0020D0, 2, 0)
      IF (STATP.NE.0) GO TO 998
      STATP = DISPTD (1, DISP, DIS_TOTDIS, 3.0000D0, 0, 0)
      IF (STATP.NE.0) GO TO 998

*     Print its contents.
      CALL FLUSH(6)
      STATP = DISPRT (1, DISP)
      IF (STATP.NE.0) GO TO 998

      WRITE (*, 70)
 70   FORMAT (/,'------------------------------------',
     :          '------------------------------------')


*   Copy it the long way first to test the various PUT and GET routines.
*     Start by reading everything from the first.
      STATP = DISGTI (1, DISP, DIS_NAXIS,  NAXIS)
      IF (STATP.NE.0) GO TO 998
      STATP = DISGTC (1, DISP, DIS_DTYPE,  DTYPE)
      IF (STATP.NE.0) GO TO 998
      STATP = DISGTI (1, DISP, DIS_NDP,    NDP)
      IF (STATP.NE.0) GO TO 998
      STATP = DISGTI (1, DISP, DIS_DP,     DP(1))
      IF (STATP.NE.0) GO TO 998
      STATP = DISGTD (1, DISP, DIS_MAXDIS, MAXDIS(1))
      IF (STATP.NE.0) GO TO 998
      STATP = DISGTD (1, DISP, DIS_TOTDIS, TOTDIS)
      IF (STATP.NE.0) GO TO 998

*     Initialize the destination.  The first argument of DISPUT (and
*     DISINI, etc.) is zero to indicate that DISL is an array (struct).
      STATL = DISPTI (0, DISL, DIS_FLAG, -1, 0, 0)
      IF (STATL.NE.0) GO TO 997
      STATL = DISINI (0, NAXIS, DISL)
      IF (STATL.NE.0) GO TO 997

*     Copy to the destination.
      JK = 0
      JN = 0
      DO 110 J = 1, NAXIS
        STATL = DISPTC (0, DISL, DIS_DTYPE, DTYPE, J, 0)
        IF (STATL.NE.0) GO TO 997
 110  CONTINUE

      DO 120 K = 1, NDP*DPLEN, DPLEN
        STATL = DISPTI (0, DISL, DIS_DP, DP(K), 0, 0)
 120  CONTINUE

      DO 130 J = 1, NAXIS
        STATL = DISPTD (0, DISL, DIS_MAXDIS, MAXDIS(J), J, 0)
        IF (STATL.NE.0) GO TO 997
 130  CONTINUE

      STATL = DISPTD (0, DISL, DIS_TOTDIS, TOTDIS, J, 0)
      IF (STATL.NE.0) GO TO 997

      STATL = DISSET (0, DISL)
      IF (STATL.NE.0) GO TO 997

*     Print the second.
      CALL FLUSH(6)
      STATL = DISPRT (0, DISL)
      IF (STATL.NE.0) GO TO 997

      WRITE (*, 70)


*   Now copy it the fast way.
*     DISCPY's first argument indicates that the source, DISP, is a
*     "pointer", and the destination, DISL, a local array.  DISCPY
*     will invoke DISINI on the destination, reusing memory allocated
*     previously by DISINI and DISSET.
      STATL = DISCPY (1, DISP, DISL)
      IF (STATL.NE.0) GO TO 997

*     A reset is required before printing.
      STATL = DISSET (0, DISL)
      IF (STATL.NE.0) GO TO 997

*     Print the copy.
      CALL FLUSH(6)
      STATL = DISPRT (0, DISL)
      IF (STATL.NE.0) GO TO 997


*   Cleanup.
 997  IF (STATL.NE.0) THEN
        STATUS = DISPERR (0, DISL, CHAR(0))
      END IF

*     Free memory allocated in the disprm struct.
      STATUS = DISFREE (0, DISL)

 998  IF (STATP.NE.0) THEN
        STATUS = DISPERR (1, DISP, CHAR(0))
      END IF

*     Free memory allocated by WCSPIH, WCSINI, etc.
 999  STATUS = WCSVFREE (NWCS, WCSP)

*     Free memory, allocated via the call to WCSSET, in the copy we made.
*     Also frees allocated memory in the shallow copy we made of linprm.
      STATUS = WCSFREE (WCS)

      END
