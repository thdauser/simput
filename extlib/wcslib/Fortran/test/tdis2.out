Testing WCSLIB Fortran disprm access routines (tdis2.f)
-------------------------------------------------------
       flag: 137
      naxis: 2
      dtype: 0x<address>
             "TPV"
             "TPV"
        ndp: 46
     ndpmax: 46
         dp: 0x<address>
               1  0            2  DQ1.NAXES
               1  0            1  DQ1.AXIS.1
               1  0            2  DQ1.AXIS.2
               1  1   0.30939     DQ1.TPV.0
               1  1   0.98842     DQ1.TPV.1
               1  1  -0.0021924   DQ1.TPV.3
               1  1   2.8381e-06  DQ1.TPV.4
               1  1   3.6848e-06  DQ1.TPV.6
               1  1   9.6423e-10  DQ1.TPV.7
               1  1   6.0581e-10  DQ1.TPV.9
               1  1  -4.0302e-09  DQ1.TPV.11
               1  1   1.7927e-12  DQ1.TPV.12
               1  1   3.5995e-12  DQ1.TPV.14
               1  1   1.7547e-12  DQ1.TPV.16
               1  1  -2.0310e-17  DQ1.TPV.17
               1  1  -2.2639e-17  DQ1.TPV.19
               1  1   3.6913e-18  DQ1.TPV.21
               1  1  -3.7771e-16  DQ1.TPV.23
               1  1   4.0081e-20  DQ1.TPV.24
               1  1   1.1800e-19  DQ1.TPV.26
               1  1   1.1810e-19  DQ1.TPV.28
               1  1   4.0114e-20  DQ1.TPV.30
               1  1   3.0321e-25  DQ1.TPV.31
               1  1   6.3743e-25  DQ1.TPV.33
               1  1   2.0538e-25  DQ1.TPV.35
               1  1  -1.9849e-25  DQ1.TPV.37
               1  1  -1.6393e-24  DQ1.TPV.39
               2  0            2  DQ2.NAXES
               2  0            2  DQ2.AXIS.1
               2  0            1  DQ2.AXIS.2
               2  1   0.99349     DQ2.TPV.1
               2  1  -1.8104e-06  DQ2.TPV.5
               2  1   4.7932e-11  DQ2.TPV.7
               2  1   6.1848e-10  DQ2.TPV.9
               2  1   1.4059e-13  DQ2.TPV.13
               2  1   8.7740e-14  DQ2.TPV.15
               2  1   2.8940e-17  DQ2.TPV.17
               2  1   1.5914e-18  DQ2.TPV.19
               2  1  -1.0280e-17  DQ2.TPV.21
               2  1  -2.8008e-21  DQ2.TPV.25
               2  1  -3.2402e-21  DQ2.TPV.27
               2  1  -7.9356e-22  DQ2.TPV.29
               2  1  -6.7738e-25  DQ2.TPV.31
               2  1  -3.0173e-25  DQ2.TPV.33
               2  1   2.8026e-25  DQ2.TPV.35
               2  1   1.4550e-25  DQ2.TPV.37
     maxdis: 0x<address>
               2.0010       2.0020    
     totdis:   3.0000    
     docorr: 0x<address>
                 0     0
       Nhat: 0x<address>
                 2     2
      axmap: 0x<address>
 axmap[0][]:     0     1
 axmap[1][]:     1     0
     offset: 0x<address>
offset[0][]:   0.0000       0.0000    
offset[1][]:   0.0000       0.0000    
      scale: 0x<address>
 scale[0][]:   1.0000       1.0000    
 scale[1][]:   1.0000       1.0000    
      iparm: 0x<address>
 iparm[0]  : 0x<address>
 iparm[0][]:            1            7           40           40            0
                        0            1
 iparm[1]  : 0x<address>
 iparm[1][]:            1            7           40           40            0
                        0            0
      dparm: 0x<address>
 dparm[0]  : 0x<address>
 dparm[0][]:   0.30939      0.98842      0.0000      -0.0021924    2.8381e-06
               0.0000       3.6848e-06   9.6423e-10   0.0000       6.0581e-10
               0.0000      -4.0302e-09   1.7927e-12   0.0000       3.5995e-12
               0.0000       1.7547e-12  -2.0310e-17   0.0000      -2.2639e-17
               0.0000       3.6913e-18   0.0000      -3.7771e-16   4.0081e-20
               0.0000       1.1800e-19   0.0000       1.1810e-19   0.0000    
               4.0114e-20   3.0321e-25   0.0000       6.3743e-25   0.0000    
               2.0538e-25   0.0000      -1.9849e-25   0.0000      -1.6393e-24
 dparm[1]  : 0x<address>
 dparm[1][]:   0.0000       0.99349      0.0000       0.0000       0.0000    
              -1.8104e-06   0.0000       4.7932e-11   0.0000       6.1848e-10
               0.0000       0.0000       0.0000       1.4059e-13   0.0000    
               8.7740e-14   0.0000       2.8940e-17   0.0000       1.5914e-18
               0.0000      -1.0280e-17   0.0000       0.0000       0.0000    
              -2.8008e-21   0.0000      -3.2402e-21   0.0000      -7.9356e-22
               0.0000      -6.7738e-25   0.0000      -3.0173e-25   0.0000    
               2.8026e-25   0.0000       1.4550e-25   0.0000       0.0000    
    i_naxis: 2
       ndis: 2
        err: 0x0
     disp2x: 0x<address>
  disp2x[0]: 0x<address>  (= tpd7)
  disp2x[1]: 0x<address>  (= tpd7)
     disx2p: 0x<address>
  disx2p[0]: 0x0
  disx2p[1]: 0x0
     tmpmem: 0x<address>
     m_flag: 137
    m_naxis: 2
    m_dtype: 0x<address>  (= dtype)
       m_dp: 0x<address>  (= dp)
   m_maxdis: 0x<address>  (= maxdis)

------------------------------------------------------------------------
       flag: 137
      naxis: 2
      dtype: 0x<address>
             "TPV"
             "TPV"
        ndp: 46
     ndpmax: 256
         dp: 0x<address>
               1  0            2  DQ1.NAXES
               1  0            1  DQ1.AXIS.1
               1  0            2  DQ1.AXIS.2
               1  1   0.30939     DQ1.TPV.0
               1  1   0.98842     DQ1.TPV.1
               1  1  -0.0021924   DQ1.TPV.3
               1  1   2.8381e-06  DQ1.TPV.4
               1  1   3.6848e-06  DQ1.TPV.6
               1  1   9.6423e-10  DQ1.TPV.7
               1  1   6.0581e-10  DQ1.TPV.9
               1  1  -4.0302e-09  DQ1.TPV.11
               1  1   1.7927e-12  DQ1.TPV.12
               1  1   3.5995e-12  DQ1.TPV.14
               1  1   1.7547e-12  DQ1.TPV.16
               1  1  -2.0310e-17  DQ1.TPV.17
               1  1  -2.2639e-17  DQ1.TPV.19
               1  1   3.6913e-18  DQ1.TPV.21
               1  1  -3.7771e-16  DQ1.TPV.23
               1  1   4.0081e-20  DQ1.TPV.24
               1  1   1.1800e-19  DQ1.TPV.26
               1  1   1.1810e-19  DQ1.TPV.28
               1  1   4.0114e-20  DQ1.TPV.30
               1  1   3.0321e-25  DQ1.TPV.31
               1  1   6.3743e-25  DQ1.TPV.33
               1  1   2.0538e-25  DQ1.TPV.35
               1  1  -1.9849e-25  DQ1.TPV.37
               1  1  -1.6393e-24  DQ1.TPV.39
               2  0            2  DQ2.NAXES
               2  0            2  DQ2.AXIS.1
               2  0            1  DQ2.AXIS.2
               2  1   0.99349     DQ2.TPV.1
               2  1  -1.8104e-06  DQ2.TPV.5
               2  1   4.7932e-11  DQ2.TPV.7
               2  1   6.1848e-10  DQ2.TPV.9
               2  1   1.4059e-13  DQ2.TPV.13
               2  1   8.7740e-14  DQ2.TPV.15
               2  1   2.8940e-17  DQ2.TPV.17
               2  1   1.5914e-18  DQ2.TPV.19
               2  1  -1.0280e-17  DQ2.TPV.21
               2  1  -2.8008e-21  DQ2.TPV.25
               2  1  -3.2402e-21  DQ2.TPV.27
               2  1  -7.9356e-22  DQ2.TPV.29
               2  1  -6.7738e-25  DQ2.TPV.31
               2  1  -3.0173e-25  DQ2.TPV.33
               2  1   2.8026e-25  DQ2.TPV.35
               2  1   1.4550e-25  DQ2.TPV.37
     maxdis: 0x<address>
               2.0010       2.0020    
     totdis:   3.0000    
     docorr: 0x<address>
                 0     0
       Nhat: 0x<address>
                 2     2
      axmap: 0x<address>
 axmap[0][]:     0     1
 axmap[1][]:     1     0
     offset: 0x<address>
offset[0][]:   0.0000       0.0000    
offset[1][]:   0.0000       0.0000    
      scale: 0x<address>
 scale[0][]:   1.0000       1.0000    
 scale[1][]:   1.0000       1.0000    
      iparm: 0x<address>
 iparm[0]  : 0x<address>
 iparm[0][]:            1            7           40           40            0
                        0            1
 iparm[1]  : 0x<address>
 iparm[1][]:            1            7           40           40            0
                        0            0
      dparm: 0x<address>
 dparm[0]  : 0x<address>
 dparm[0][]:   0.30939      0.98842      0.0000      -0.0021924    2.8381e-06
               0.0000       3.6848e-06   9.6423e-10   0.0000       6.0581e-10
               0.0000      -4.0302e-09   1.7927e-12   0.0000       3.5995e-12
               0.0000       1.7547e-12  -2.0310e-17   0.0000      -2.2639e-17
               0.0000       3.6913e-18   0.0000      -3.7771e-16   4.0081e-20
               0.0000       1.1800e-19   0.0000       1.1810e-19   0.0000    
               4.0114e-20   3.0321e-25   0.0000       6.3743e-25   0.0000    
               2.0538e-25   0.0000      -1.9849e-25   0.0000      -1.6393e-24
 dparm[1]  : 0x<address>
 dparm[1][]:   0.0000       0.99349      0.0000       0.0000       0.0000    
              -1.8104e-06   0.0000       4.7932e-11   0.0000       6.1848e-10
               0.0000       0.0000       0.0000       1.4059e-13   0.0000    
               8.7740e-14   0.0000       2.8940e-17   0.0000       1.5914e-18
               0.0000      -1.0280e-17   0.0000       0.0000       0.0000    
              -2.8008e-21   0.0000      -3.2402e-21   0.0000      -7.9356e-22
               0.0000      -6.7738e-25   0.0000      -3.0173e-25   0.0000    
               2.8026e-25   0.0000       1.4550e-25   0.0000       0.0000    
    i_naxis: 2
       ndis: 2
        err: 0x0
     disp2x: 0x<address>
  disp2x[0]: 0x<address>  (= tpd7)
  disp2x[1]: 0x<address>  (= tpd7)
     disx2p: 0x<address>
  disx2p[0]: 0x0
  disx2p[1]: 0x0
     tmpmem: 0x<address>
     m_flag: 137
    m_naxis: 2
    m_dtype: 0x<address>  (= dtype)
       m_dp: 0x<address>  (= dp)
   m_maxdis: 0x<address>  (= maxdis)

------------------------------------------------------------------------
       flag: 137
      naxis: 2
      dtype: 0x<address>
             "TPV"
             "TPV"
        ndp: 46
     ndpmax: 256
         dp: 0x<address>
               1  0            2  DQ1.NAXES
               1  0            1  DQ1.AXIS.1
               1  0            2  DQ1.AXIS.2
               1  1   0.30939     DQ1.TPV.0
               1  1   0.98842     DQ1.TPV.1
               1  1  -0.0021924   DQ1.TPV.3
               1  1   2.8381e-06  DQ1.TPV.4
               1  1   3.6848e-06  DQ1.TPV.6
               1  1   9.6423e-10  DQ1.TPV.7
               1  1   6.0581e-10  DQ1.TPV.9
               1  1  -4.0302e-09  DQ1.TPV.11
               1  1   1.7927e-12  DQ1.TPV.12
               1  1   3.5995e-12  DQ1.TPV.14
               1  1   1.7547e-12  DQ1.TPV.16
               1  1  -2.0310e-17  DQ1.TPV.17
               1  1  -2.2639e-17  DQ1.TPV.19
               1  1   3.6913e-18  DQ1.TPV.21
               1  1  -3.7771e-16  DQ1.TPV.23
               1  1   4.0081e-20  DQ1.TPV.24
               1  1   1.1800e-19  DQ1.TPV.26
               1  1   1.1810e-19  DQ1.TPV.28
               1  1   4.0114e-20  DQ1.TPV.30
               1  1   3.0321e-25  DQ1.TPV.31
               1  1   6.3743e-25  DQ1.TPV.33
               1  1   2.0538e-25  DQ1.TPV.35
               1  1  -1.9849e-25  DQ1.TPV.37
               1  1  -1.6393e-24  DQ1.TPV.39
               2  0            2  DQ2.NAXES
               2  0            2  DQ2.AXIS.1
               2  0            1  DQ2.AXIS.2
               2  1   0.99349     DQ2.TPV.1
               2  1  -1.8104e-06  DQ2.TPV.5
               2  1   4.7932e-11  DQ2.TPV.7
               2  1   6.1848e-10  DQ2.TPV.9
               2  1   1.4059e-13  DQ2.TPV.13
               2  1   8.7740e-14  DQ2.TPV.15
               2  1   2.8940e-17  DQ2.TPV.17
               2  1   1.5914e-18  DQ2.TPV.19
               2  1  -1.0280e-17  DQ2.TPV.21
               2  1  -2.8008e-21  DQ2.TPV.25
               2  1  -3.2402e-21  DQ2.TPV.27
               2  1  -7.9356e-22  DQ2.TPV.29
               2  1  -6.7738e-25  DQ2.TPV.31
               2  1  -3.0173e-25  DQ2.TPV.33
               2  1   2.8026e-25  DQ2.TPV.35
               2  1   1.4550e-25  DQ2.TPV.37
     maxdis: 0x<address>
               2.0010       2.0020    
     totdis:   3.0000    
     docorr: 0x<address>
                 0     0
       Nhat: 0x<address>
                 2     2
      axmap: 0x<address>
 axmap[0][]:     0     1
 axmap[1][]:     1     0
     offset: 0x<address>
offset[0][]:   0.0000       0.0000    
offset[1][]:   0.0000       0.0000    
      scale: 0x<address>
 scale[0][]:   1.0000       1.0000    
 scale[1][]:   1.0000       1.0000    
      iparm: 0x<address>
 iparm[0]  : 0x<address>
 iparm[0][]:            1            7           40           40            0
                        0            1
 iparm[1]  : 0x<address>
 iparm[1][]:            1            7           40           40            0
                        0            0
      dparm: 0x<address>
 dparm[0]  : 0x<address>
 dparm[0][]:   0.30939      0.98842      0.0000      -0.0021924    2.8381e-06
               0.0000       3.6848e-06   9.6423e-10   0.0000       6.0581e-10
               0.0000      -4.0302e-09   1.7927e-12   0.0000       3.5995e-12
               0.0000       1.7547e-12  -2.0310e-17   0.0000      -2.2639e-17
               0.0000       3.6913e-18   0.0000      -3.7771e-16   4.0081e-20
               0.0000       1.1800e-19   0.0000       1.1810e-19   0.0000    
               4.0114e-20   3.0321e-25   0.0000       6.3743e-25   0.0000    
               2.0538e-25   0.0000      -1.9849e-25   0.0000      -1.6393e-24
 dparm[1]  : 0x<address>
 dparm[1][]:   0.0000       0.99349      0.0000       0.0000       0.0000    
              -1.8104e-06   0.0000       4.7932e-11   0.0000       6.1848e-10
               0.0000       0.0000       0.0000       1.4059e-13   0.0000    
               8.7740e-14   0.0000       2.8940e-17   0.0000       1.5914e-18
               0.0000      -1.0280e-17   0.0000       0.0000       0.0000    
              -2.8008e-21   0.0000      -3.2402e-21   0.0000      -7.9356e-22
               0.0000      -6.7738e-25   0.0000      -3.0173e-25   0.0000    
               2.8026e-25   0.0000       1.4550e-25   0.0000       0.0000    
    i_naxis: 2
       ndis: 2
        err: 0x0
     disp2x: 0x<address>
  disp2x[0]: 0x<address>  (= tpd7)
  disp2x[1]: 0x<address>  (= tpd7)
     disx2p: 0x<address>
  disx2p[0]: 0x0
  disx2p[1]: 0x0
     tmpmem: 0x<address>
     m_flag: 137
    m_naxis: 2
    m_dtype: 0x<address>  (= dtype)
       m_dp: 0x<address>  (= dp)
   m_maxdis: 0x<address>  (= maxdis)
